fairlearn/metrics/_metric_frame.py

```diff
@@ -382,19 +382,41 @@ def group_max(self) -> Union[Any, pd.Series, pd.DataFrame]:
         whether control features are present, and whether the metric functions
         were specified as a single callable or a dictionary.
 
+        Parameters
+        ----------
+        errors: {'ignore, 'raise', 'coerce'}, default 'raise'
+            if 'raise', then invalid parsing will raise an exception
+            if 'coerce', then invalid parsing will be set as NaN
+            if 'ignore', then invalid parsing will return the input
+
         Returns
         -------
         typing.Any or pandas.Series or pandas.DataFrame
             The maximum value over sensitive features. The exact type
             follows the table in :attr:`.MetricFrame.overall`.
         """
-        if not self.control_levels:
-            result = pd.Series(index=self._by_group.columns, dtype='object')
-            for m in result.index:
-                max_val = self._by_group[m].max()
-                result[m] = max_val
-        else:
-            result = self._by_group.groupby(level=self.control_levels).max()
+        if errors not in ("ignore", "raise", "coerce"):
+            raise ValueError("invalid error value specified")
+
+        try:
+            if not self.control_levels:
+                result = pd.Series(index=self._by_group.columns, dtype='object')
+                for m in result.index:
+                    max_val = self._by_group[m].max()
+                    result[m] = max_val
+            else:
+                result = self._by_group.groupby(level=self.control_levels).max()
+        # Catch ValueError that is thrown by pandas, if not all columns in metric frame are scalar
+        except ValueError:
+            warnings.warn(f"Metric frame contains non-scalar cells. "
+                          f"Please remove non-scalar columns from your metric frame and try again",
+                          SyntaxWarning)
+            if errors == "raise":
+                raise
+            elif errors == 'coerce':
+                return np.nan
+            else:
+                return self._by_group
```

Comment:
I like this a lot. However, I'm not 100% sure I understand when I'd want to have it return `self._by_group`. Can you imagine a situation where that's useful?

Reply:
Good question. I *could* imagine that if you prepare a generic pipeline that could deal with metrics for which difference makes sense and metrics for which it doesn't (e.g., a confusion matrix), this could be useful. But that's probably such a niche scenario that we might be better off limiting the options to 'coerce' and 'raise'. 

WDYT?

Reply:
I thought it would be useful since the warning message says "some columns contain non-scalar cells", that the user gets to see the columns as output. But I do not see any other usecase then that.

Reply:
Shall I change it to only have 'raise' and 'coerce'? And then have 'coerce' as default for .difference() and .ratio()?

Reply:
And (as alluded above) what do you do when there's one column which is a bunch of floats, and another column which consists of confusion matrices?

Reply:
@riedgar-ms I'm not entirely sure if you mean that as an argument for or against having this third option 🙂 
If we have a metric for which difference (or ratio, etc.) doesn't make sense, then why are we returning `self._by_group`? We could return `np.nan` perhaps but then that's the `coerce` path.

Reply:
@riedgar-ms can you comment?

Reply:
I was thinking that if you had a mixture of return types where some of them did not have 'less than' well defined, then calling with `raise` would raise an exception, and `coerce` would return the minimums for columns where that was defined, and NAN for columns where a minimum is not defined. Does that answer your question @romanlutz ?